server/image_fs_info.go

```diff
@@ -2,12 +2,130 @@ package server
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"syscall"
 	"time"
 
+	"github.com/containers/storage"
+	"github.com/docker/docker/pkg/mount"
 	"golang.org/x/net/context"
+	"golang.org/x/sys/unix"
 	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
 )
 
+func getStorageFsInfo(store storage.Store) (*pb.FilesystemUsage, error) {
+	rootPath := store.GraphRoot()
+	storageDriver := store.GraphDriverName()
+	filePath := path.Join(rootPath, storageDriver+"-images")
+
+	statInfo := syscall.Stat_t{}
+	err := syscall.Lstat(filePath, &statInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	mounts, err := mount.GetMounts()
+	if err != nil {
+		return nil, err
+	}
+
+	deviceName, err := getDeviceName(statInfo, mounts)
+	if err != nil {
+		return nil, err
+	}
+
+	uuid, err := getDeviceToUUID(deviceName)
+	if err != nil {
+		return nil, err
+	}
+
+	bytesUsed, inodesUsed, err := getDiskUsageStats(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	usage := pb.FilesystemUsage{
+		Timestamp:  time.Now().UnixNano(),
+		StorageId:  &pb.StorageIdentifier{uuid},
+		UsedBytes:  &pb.UInt64Value{bytesUsed},
+		InodesUsed: &pb.UInt64Value{inodesUsed},
+	}
+
+	return &usage, nil
+}
+
+func getDeviceToUUID(devicePath string) (string, error) {
+	const dir = "/dev/disk/by-uuid"
+
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		return "", nil
+	}
+
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return "", err
+	}
+
+	for _, file := range files {
+		path := filepath.Join(dir, file.Name())
+		target, err := os.Readlink(path)
+		if err != nil {
+			continue
+		}
+		device, err := filepath.Abs(filepath.Join(dir, target))
+		if err != nil {
+			return "", fmt.Errorf("failed to resolve the absolute path of %q", filepath.Join(dir, target))
+		}
+		if strings.Compare(device, devicePath) == 0 {
+			return file.Name(), nil
+		}
+	}
+
+	return "", fmt.Errorf("Device path not found")
```

Comment:
nit: Returned errors should start with lower case and should also include the devicePath in the error message here:
```
return "", fmt.Errorf("device path %s not found", devicePath)
```